things/groups.go

```diff
@@ -0,0 +1,168 @@
+package things
+
+import (
+	"context"
+)
+
+type Group interface {
+	ID() string
+	OwnerID() string
+	ParentID() string
+	Name() string
+	Description() string
+	Metadata() Metadata
+
+	SetID(id string)
```

Comment:
Do we really need all these setters? Why not passing group struct pointer and changing the value directly.

This looks like some OO encapsulation, maybe it is needed but it Go it gives me "code smell".

Reply:
Idea here and in `things/http.go` is to extract the common code that can be used for other entities that we wish to group
so `Group` is an interface that would have its implementations in each entity package itself
Of course if this solution is ok, http.go and `groups.go` would be moved to appropriate package

Reply:
This is OK, but I do not like these setters. You can create an var struct without the setters, and then change internal struct directly through pointer receiver.